ext/phar/phar_object.c

```diff
@@ -4417,19 +4417,40 @@ PHP_METHOD(Phar, extractTo)
 		phar_archive_data *phar;
 all_files:
 		phar = phar_obj->archive;
-		/* Extract all files */
+		/* Extract all files, or if we have jumped here from a potential
+		   directory match, just those manifest entries matching the dir */
 		if (!zend_hash_num_elements(&(phar->manifest))) {
```

Comment:
This check may prevent an exception from being thrown. It should be fine to just drop this check.